### PR TITLE
AudioSource SDKComponent

### PR DESCRIPTION
**Pull request by popuz**
Add support for audio (sources) in SDK scenes.
- downloading, assigning and clean-up of the audio clips and respective audio sources
- budgeted and cached with supported cache clean-up
- add profiling markers for AudioClips
- unit, integration and one automated (big integration with real sdk scene) tests

# How to test
- start game and go to streaming-world
- switch to first-person to have proper audio spatialization
- verify you hear music
- run to dance floor
image
- verify you hear dance-floor music

### Note
- music not-playing outside of the scene is not implemented yet
- proper audio spatialization works stable in first person for now (because of how our cameras works)

### CODE REVIEW

**Comment by anicalbano** [comment]
✔️ Check the first iteration of the PR and looks correct. 
A comment regarding the sound performance is that, if we change realms after being inside the streaming world and then go back, the audio stops.
The video attached sounds low because of my own recording, but in world it sounds correct.

https://github.com/decentraland/unity-explorer/assets/139157886/ecb1f7e5-9a38-4a34-9058-16de156a1946

**Review by mikhail-dcl — CHANGES_REQUESTED** [review]
The flow looks absolutely correct but I found several things requiring a fix

File: Explorer/Assets/DCL/PluginSystem/World/AudioSourcesPlugin.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+        public AudioSourcesPlugin(ECSWorldSingletonSharedDependencies sharedDependencies, IWebRequestController webRequestController, CacheCleaner cacheCleaner)
+        {
+            this.webRequestController = webRequestController;
+
+            frameTimeBudgetProvider = sharedDependencies.FrameTimeBudget;
+            memoryBudgetProvider = sharedDependencies.MemoryBudget;
+
+            componentPoolsRegistry = sharedDependencies.ComponentPoolsRegistry;
+            componentPoolsRegistry.AddGameObjectPool<AudioSource>();
```
[the remark this answers is not in the document]
Comment (popuz) [reply Explorer/Assets/DCL/PluginSystem/World/AudioSourcesPlugin.cs:32]: I was thinking that this logic is handled in `CleanUpAudioSourceSystem` via `ReleaseOnEntityDestroy` (that also handles component removal). And `clip` nullifying is inside `AudioSourceComponent.Dispose`. As I checked, it works correctly now - released AudioSources inside pool have nothing assigned in the `clip` inspector field. 
Do I still need to nullify them here?

[the remark this answers is not in the document]
Comment (mikhail-dcl) [reply Explorer/Assets/DCL/PluginSystem/World/AudioSourcesPlugin.cs]:
```
if (poolsRegistry.TryGetPool(typeof(AudioSource), out IComponentPool componentPool))
                    componentPool.Release(component.Result);

                component.Dispose();
```
It's a bit obscure as dispose is called after releasing to the pool but in general it's fine.
I just don't expect that that the component is being manipulated **after** its internals are released to the pool

File: Explorer/Assets/DCL/SDKComponents/AudioSources/AudioUtils.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            if (!string.IsNullOrEmpty(url))
+                return url[^3..].ToLower() switch
+                       {
+                           "mp3" => AudioType.MPEG,
+                           "wav" => AudioType.WAV,
+                           "ogg" => AudioType.OGGVORBIS,
+                           _ => AudioType.UNKNOWN,
+                       };
+
+            Debug.LogError($"Cannot detect AudioType. UrlName doesn't contain file extension!. Setting to {AudioType.UNKNOWN.ToString()}");
```
[the remark this answers is not in the document]
Comment (popuz) [reply Explorer/Assets/DCL/SDKComponents/AudioSources/AudioUtils.cs]: added ValidationTest to also prevent it in other classes. For now this test excludes `JsonUtils.cs`, `WorldSyncCommandBufferCollectionsPool.cs` and `Sentry` folder

**Review by mikhail-dcl — APPROVED** [review]

**Pull request merged**